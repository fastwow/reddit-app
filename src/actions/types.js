export const FETCH_POSTS_STARTED = 'fetch_posts_started';
export const FETCH_POSTS_SUCCESS = 'fetch_posts_success';
export const FETCH_MORE_POSTS_STARTED = 'fetch_more_posts_started';
export const FETCH_MORE_POSTS_SUCCESS = 'fetch_more_posts_success';
export const FETCH_REFRESHED_POSTS_STARTED = 'fetch_refreshed_posts_started';
export const FETCH_REFRESHED_POSTS_SUCCESS = 'fetch_refreshed_posts_success';
export const FETCH_FAVORITES_POSTS_STARTED = 'fetch_favorites_posts_started';
export const FETCH_FAVORITES_POSTS_SUCCESS = 'fetch_favorites_posts_success';
export const ADD_TO_FAVORITE_SUCCESS = 'add_to_favorite_success';
export const REMOVE_FROM_FAVORITE_SUCCESS = 'remove_from_favorite_success';
export const FILTER_POSTS_SUCCESS = 'search_post_success';
