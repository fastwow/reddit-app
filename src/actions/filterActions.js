import {FILTER_POSTS} from './types';

export function filter(searchTerm) {
  return {
    type: FILTER_POSTS,
    searchTerm,
  };
}
