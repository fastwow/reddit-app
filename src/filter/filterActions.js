import {FILTER_POSTS} from './filterActionTypes';

export function filter(searchTerm) {
  return {
    type: FILTER_POSTS,
    searchTerm,
  };
}
