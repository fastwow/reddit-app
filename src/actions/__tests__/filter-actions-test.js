import {filter} from '../filterActions';
import {FILTER_POSTS} from '../types';

it(`creates ${FILTER_POSTS} action`, () => {
  expect(filter('cat')).toMatchSnapshot();
});
