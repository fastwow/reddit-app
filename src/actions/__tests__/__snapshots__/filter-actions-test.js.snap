// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates filter_posts action 1`] = `
Object {
  "searchTerm": "cat",
  "type": "filter_posts",
}
`;
